src/Chapter4.hs

```diff
@@ -611,6 +647,10 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+instance Monad List where
+    (>>=) :: List a -> (a -> List b) -> List b
+    Empty >>= _ = Empty
+    lst >>= fun = listFoldl concatList Empty (mapList fun lst)
```

Comment:
Here you also can use `fmap`, and then you will need list inlining (traditionally named via concat smth in Haskell).